src/index.rs

```diff
@@ -208,31 +229,59 @@ impl std::fmt::Display for TableId {
 }
 
 impl IndexTable {
-	pub fn open_existing(path: &std::path::Path, id: TableId) -> Result<Option<IndexTable>> {
-		let mut path: std::path::PathBuf = path.into();
-		path.push(id.file_name());
-
-		let file = match std::fs::OpenOptions::new().read(true).write(true).open(path.as_path()) {
-			Err(e) if e.kind() == std::io::ErrorKind::NotFound => return Ok(None),
-			Err(e) => return Err(Error::Io(e)),
-			Ok(file) => file,
-		};
+	pub fn open_existing(
+		path: &std::path::Path,
+		id: TableId,
+		max_size: Option<usize>,
+	) -> Result<Option<IndexTable>> {
+		let path_base: std::path::PathBuf = path.into();
+		let mut maps = Vec::new();
+		for i in 0.. {
+			let mut path = path_base.clone();
+			path.push(id.file_name(max_size.is_some().then(|| i)));
+
+			let file = match std::fs::OpenOptions::new().read(true).write(true).open(path.as_path())
+			{
+				Err(e) if e.kind() == std::io::ErrorKind::NotFound =>
+					if i == 0 {
+						return Ok(None)
+					} else {
+						break
+					},
+				Err(e) => return Err(Error::Io(e)),
+				Ok(file) => file,
+			};
 
-		try_io!(file.set_len(file_size(id.index_bits())));
-		let mut map = try_io!(unsafe { memmap2::MmapMut::map_mut(&file) });
-		madvise_random(&mut map);
-		log::debug!(target: "parity-db", "Opened existing index {}", id);
-		Ok(Some(IndexTable { id, path, map: RwLock::new(Some(map)) }))
+			try_io!(file.set_len(file_size(id.index_bits())));
```

Comment:
Looks like all the index files are getting the original total size rather than smaller size based on max_chunks.

Reply:
great catch, thanks :+1: